File: codegen/results/v002/ollama/qwen3.6_35b_nothink/20260418_170014/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

function countWordFrequency(filePath: string): void {
    const content = fs.readFileSync(filePath, 'utf-8');
    const words = content.toLowerCase().split(/\s+/);
    
    const frequency: Record<string, number> = {};
    
    for (const rawWord of words) {
        // Strip punctuation: keep only letters
        const cleanWord = rawWord.replace(/[^a-z]/g, '');
        if (cleanWord.length === 0) {
            continue;
        }
        if (!frequency[cleanWord]) {
            frequency[cleanWord] = 0;
        }
        frequency[cleanWord]++;
    }
    
    // Convert to array for sorting
    const entries = Object.entries(frequency);
    
    // Sort by count descending, then alphabetically ascending for ties
    entries.sort((a, b) => {
        if (b[1] !== a[1]) {
            return b[1] - a[1];
        }
        return a[0].localeCompare(b[0]);
    });
    
    // Output
    for (const [word, count] of entries) {
        console.log(`${word}: ${count}`);
    }
}

const inputPath = path.join('input', 'text.txt');
countWordFrequency(inputPath);

File: codegen/results/v002/ollama/qwen3.6_35b_nothink/20260418_170014/python/002_word_frequency/solution.py
import re
from collections import Counter

# Read the file
with open('input/text.txt', 'r') as f:
    text = f.read()

# Convert to lowercase
text = text.lower()

# Find all words (sequences of letters only)
words = re.findall(r'[a-z]+', text)

# Count frequency
counter = Counter(words)

# Sort by count descending, then alphabetically ascending
sorted_words = sorted(counter.items(), key=lambda x: (-x[1], x[0]))

# Output
for word, count in sorted_words:
    print(f"{word}: {count}")